clinica/pipelines/t1_linear/t1_linear_utils.py

```diff
@@ -14,7 +13,24 @@ def get_substitutions_datasink(bids_file):
             f"{bids_file}_space-MNI152NLin2009cSym_res-1x1x1_T1w.nii.gz",
         ),
     ]
-    return bids_file, substitutions_ls
+    substitutions_ls_flair = [  # registration
+        (
+            f"{bids_file}Warped_cropped.nii.gz",
+            f"{bids_file}_space-MNI152NLin2009cSym_desc-Crop_res-1x1x1_flair.nii.gz",
+        ),
+        (
+            f"{bids_file}0GenericAffine.mat",
+            f"{bids_file}_space-MNI152NLin2009cSym_res-1x1x1_affine.mat",
+        ),
+        (
+            f"{bids_file}Warped.nii.gz",
+            f"{bids_file}_space-MNI152NLin2009cSym_res-1x1x1_flair.nii.gz",
+        ),
+    ]
+    if pipeline_name == "t1-linear":
+        return bids_file, substitutions_ls_t1
+    else:
+        return bids_file, substitutions_ls_flair
```

Comment:
This function can be simplified a lot. Define the substitution list only once but use variables to format the names:

```python
suffix = "T1w" if pipeline_name == "t1-linear" else "flair"
substitutions = [
    (
             f"{bids_file}Warped_cropped.nii.gz",
             f"{bids_file}_space-MNI152NLin2009cSym_desc-Crop_res-1x1x1_{suffix}.nii.gz",
         ),
         (
             f"{bids_file}0GenericAffine.mat",
             f"{bids_file}_space-MNI152NLin2009cSym_res-1x1x1_affine.mat",
         ),
         (
             f"{bids_file}Warped.nii.gz",
             f"{bids_file}_space-MNI152NLin2009cSym_res-1x1x1_{suffix}.nii.gz",
         ),
     ]
```

Besides, the function has `bids_file` in the input and output without doing anything to it --> it should be removed from the outputs.